architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,206 @@
+      host: "0.0.0.0"
+      port: 5443
+    workerServer:
+      maxOperationConcurrency: 10
+      maxOperationRetryCount: 2
+    ucp:
+      kind: kubernetes
+    logging:
+      level: "info"
+      json: true
+    {{- if and .Values.global.zipkin .Values.global.zipkin.url }}
+    tracerProvider:
+      serviceName: "applications.core"
+      zipkin: 
+        url: {{ .Values.global.zipkin.url }}
+    {{- end }}
+```
+
+#### Use of Cryptography
+
+#### Storage of secrets
+
+Applications RP stores secrets for rendering some resources. For instance TLS termination in gateways needs storing TLS cert and key. For these cases, Applications RP uses kubernetes secrets. 
+
+#### Data Serialization / Formats
+
+### Clients
+
+**UCP** is the only client of Applications RP. It communicates with Applications RP for deploying various Radius Application resources. 
+
+All communications use mTLS.
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the Applications RP component:
+
+- **Kubernetes Cluster**: The overall environment where the Applications RP  component operates and serves clients.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads. 
+
+The Applications RP component lives inside the `radius-system` namespace in the Kubernetes cluster where it is installed. UCP and Controller also reside within the same namespace. Namespaces within Kubernetes can help set Role-Based Access Control (RBAC) policies.
+
+Applications RP component communicates with Kube API server through Kube Controller for saving Radius resources and for queuing async operations. The API Server and Kube Controller run in the `kube-system` namespace within the cluster. 
+
+## Assumptions
+
+This threat model assumes that:
+
+1. The Radius installation is not tampered with.
+2. The Kubernetes cluster that Radius is installed on is not compromised.
+3. It is the responsibility of the Kubernetes cluster to authenticate users. Administrators and users with sufficient privileges can perform their required tasks. Radius cannot prevent actions taken by an administrator.
+
+## Data Flow
+
+### Diagram
+
+![Applications RP via Microsoft Threat Modeling Tool](2024-10-applications-rp-threat-model/apprp-dataflow.png)
+
+Below are the key points associated with data flow:
+1. Applications RP receives request to dpeloy resources from UCP and sends back appropriate response.
+2. Applications RP communicates with Controller for managing kubernetes resources and validating custom recipes.
+3. Applications RP requests UCP to deploy bicep recipes.
+4.  Application RP (terraform provider) requests AWS/ Azure to deploy resources for terraform recipes.
+5.  Application RP uses API server through Kubernetes Controller to save Radius resources and Async Operations ( APIServer is used as datastore and Queue).
+6.  Application RP fetches recipes from OCI registries and public terraform modules.
+
+
+### Threats
```

Comment:
I'm going to hold off on reviewing threats until the first part of the document is done.

Reply:
we finished this.